pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -98,14 +93,37 @@
     };
   }
 
+  /**
+   * Helper method to construct a OrderByValueExtractor based on the given expression.
+   */
+  private OrderByValueExtractor getOrderByValueExtractor(ExpressionContext expression) {
+    if (expression.getType() == ExpressionContext.Type.LITERAL) {
```

Comment:
what does it mean to have ORDER BY a literal? can you give an example?

Reply:
Order-by with literal itself does not make a lot of sense, but post-aggregation might contain literal (e.g. `ORDER BY SUM(col1) * 2 - SUM(col2)`)